plugins/module_utils/network/ios/config/vrf_global/vrf_global.py

```diff
@@ -0,0 +1,123 @@
+#
+# -*- coding: utf-8 -*-
+# Copyright 2024 Red Hat
+# GNU General Public License v3.0+
+# (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+#
+
+from __future__ import absolute_import, division, print_function
+
+
+__metaclass__ = type
+
+"""
+The ios_vrf_global config file.
+It is in this file where the current configuration (as dict)
+is compared to the provided configuration (as dict) and the command set
+necessary to bring the current configuration to its desired end-state is
+created.
+"""
+
+from copy import deepcopy
+
+from ansible.module_utils.six import iteritems
+from ansible_collections.ansible.netcommon.plugins.module_utils.network.common.rm_base.resource_module import (
+    ResourceModule,
+)
+from ansible_collections.ansible.netcommon.plugins.module_utils.network.common.utils import (
+    dict_merge,
+)
+
+from ansible_collections.cisco.ios.plugins.module_utils.network.ios.facts.facts import Facts
+from ansible_collections.cisco.ios.plugins.module_utils.network.ios.rm_templates.vrf_global import (
+    Vrf_globalTemplate,
+)
+
+
+class Vrf_global(ResourceModule):
+    """
+    The ios_vrf_global config class
+    """
+
+    def __init__(self, module):
+        super(Vrf_global, self).__init__(
+            empty_fact_val={},
+            facts_module=Facts(module),
+            module=module,
+            resource="vrf_global",
+            tmplt=Vrf_globalTemplate(),
+        )
+        self.parsers = [
+            "name",
+            "description",
+            "ipv4.multicast.multitopology",
+            "ipv6.multicast.multitopology",
+            "rd",
+            "route_target.export",
+            "route_target.import_config",
+            "route_target.both",
+            "vnet.tag",
+            "vpn.id",
+        ]
+
+    def execute_module(self):
+        """Execute the module
+
+        :rtype: A dictionary
+        :returns: The result from module execution
+        """
+        if self.state not in ["parsed", "gathered"]:
+            self.generate_commands()
+            self.run_commands()
+        return self.result
+
+    def generate_commands(self):
+        """Generate configuration commands to send based on
+        want, have and desired state.
+        """
+        wantd = {entry["name"]: entry for entry in self.want}
+        haved = {entry["name"]: entry for entry in self.have}
+
+        # if state is merged, merge want onto have and then compare
+        if self.state == "merged":
+            wantd = dict_merge(haved, wantd)
+
+        # if state is deleted, empty out wantd and set haved to wantd
+        if self.state == "deleted":
+            haved = {k: v for k, v in iteritems(haved) if k in wantd or not wantd}
+            wantd = {}
+
+        if self.state == "purged":
```

Comment:
Purged would delete things as a whole, adding ref config below -
```
Router3#conf t
Enter configuration commands, one per line.  End with CNTL/Z.
Router3(config)#no vrf definition test  <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<< PURGE STATE
% IPv4 and IPv6 addresses from all interfaces in VRF test have been removed
Router3(config)#end
Router3#sh running-config | section vrf
vrf definition ansible

```

Reply:
address-family configuration comes under each vrfs configuration so if a vrf definition is negated the address-family is also gone. I believe we need not take any specific steps to delete the address family.
`no vrf definition <vrf name>` should be enough